### PR TITLE
Allow for overriding the install dir on a per-edition level

### PR DESCRIPTION
**Pull request by mattcl**

### CODE REVIEW

**Comment by zirkov** [comment]
@mattcl https://github.com/Roblox/visualstudio/blob/master/CHANGELOG.md entry?
Otherwise, LGTM

**Comment by Brantone** [comment]
Need update to CHANGELOG.md

File: recipes/install.rb [hunk 1]
```diff
@@ -26,10 +26,13 @@
   version = install['version']
   edition = install['edition']
   url = source_download_url(version, edition)
+  # allow to override the install dir on a per-edition basis
+  install_directory = node['visualstudio'][version][edition]['install_dir'] || node['visualstudio'][version]['install_dir'] # rubocop:disable Metrics/LineLength
```
Comment (Brantone) [inline recipes/install.rb:30]: This implies there could be both a `community` and `buildtools` editions installed ... how does the machine handle that when they're both 2017, won't there be overlap?

Comment (mattcl) [reply recipes/install.rb:30]: Unsure, but I know I can't install both microsoft build tools and the agents for visual studio in the same directory. Teamcity identifies the separate paths correctly. So it ends up being `../2017/BuildTools` and `../2017/TestAgent`. I verified that our builds appear to still work with both installed.

**Comment by mattcl** [comment]
@zirkov right, I keep forgetting. I'm so used to using https://github.com/saschagrunert/git-journal that I forget the changelogs won't be auto-generated.

File: recipes/install.rb (hunk 1 above)
Comment (mattcl) [reply recipes/install.rb:30]: I should also mention that I consider this situation user error.

**Comment by zirkov** [comment]
@mattcl Great tool to look for when we'll work on our local git repo, thanks! But as it requires Rust work environment I'd say it's a little bit of a stretch to require it for all commits to our Chef repos

File: recipes/install.rb (hunk 1 above)
Comment (zirkov) [reply recipes/install.rb:30]: Shouldn't then the recipe just raise an exception if _both_ are defined?

**Pull request merged**